File: avancado-intermediario/Python/Q5/veiculo.py
from objeto import Objeto
from movimento import Movimento

class Veiculo(Objeto, Movimento):

    def __init__(self, cor, x = 0.0, y = 0.0):
        super().__init__(x, y)
        self.__cor = cor

    def getCor(self):
        return self.__cor

    def mover(self, x, y):
        self.setX(x)
        self.setY(y)

    def __str__(self):
        return (f"{super().__str__()}"
                f"Cor = {self.getCor()}")

File: avancado-intermediario/Java/Q5/Veiculo.java
public class Veiculo extends Objeto implements Movimento {

    private String cor;

    public Veiculo(String cor) {
        this(cor, 0.0, 0.0);
    }

    public Veiculo(String cor, double x, double y) {
        super(x, y);
        this.cor = cor;
    }

    public String getCor() {
        return cor;
    }

    @Override
    public void mover(double x, double y) {
        setX(x);
        setY(y);
    }

    @Override
    public String toString() {
        return super.toString() +
               "Cor = " + cor;
    }
}
